### ISSUE TITLE
Install php extension with pecl on windows with version.

### ISSUE THREAD

**Issue by terabytesoftw** [shivammathur/setup-php#855]
**Describe the bug**

Install wincache-2.0.0.9

**Version**
- [x] `v2`

**Operating systems**
Windows

**PHP versions**
8.1

**Screenshots/Logs**
```
Add-Extension: D:\a\_actions\shivammathur\setup-php\main\src\scripts\run.ps1:15
Line |
  15 |  Add-Extension wincache stable 2.0.0.9
     |                                ~~~~~~~
     | Cannot validate argument on parameter 'extension_version'. The argument "2.0.0.9" does not match the
     | "^\d+(\.\d+){0,2}$" pattern. Supply an argument that matches "^\d+(\.\d+){0,2}$" and try the command again.
```

**Comment by shivammathur** [comment]
I fixed the regex here on setup-php side https://github.com/shivammathur/setup-php/commit/594bd591ec5fc2a0913e1e6c6eba2bd60844146b.

@mlocati 
Can we have it changed in `Install-PhpExtension` as well.
And, also changed to search for DLLs in https://downloads.php.net/~windows/pecl/ indexes.

**Comment by terabytesoftw** [comment]
> I fixed the regex here on setup-php side [594bd59](https://github.com/shivammathur/setup-php/commit/594bd591ec5fc2a0913e1e6c6eba2bd60844146b).
> 
> @mlocati Can we have it changed in `Install-PhpExtension` as well. And, also changed to search for DLLs in https://downloads.php.net/~windows/pecl/ indexes.

Thks, use the branch but it doesn't install it.

````
✗ wincache Could not install wincache on PHP 8.1.29 
```

**Comment by mlocati** [comment]
> Can we have it changed in `Install-PhpExtension` as well.

Done: https://github.com/mlocati/powershell-phpmanager/pull/118

**Comment by shivammathur** [comment]
@mlocati 
Thanks for updating the regex.

New extension builds are no longer uploaded to windows.php.net, Would it be possible to change it so that it looks for builds in https://downloads.php.net/~windows/pecl/

**Comment by mlocati** [comment]
If I'm not wrong, I should already have done that: see https://github.com/mlocati/powershell-phpmanager/pull/116

**Comment by mlocati** [comment]
Thanks to you!

**Comment by shivammathur** [comment]
Released [`2.31.1`](https://github.com/shivammathur/setup-php/releases/tag/2.31.1) with the fix.
